packages/opentelemetry-sdk-trace-base/src/export/BatchSpanProcessorBase.ts

```diff
@@ -236,5 +231,26 @@ export abstract class BatchSpanProcessorBase<T extends BufferConfig>
     }
   }
 
+  private _flushOneBatchAndContinue() {
+    this._flushOneBatch()
+      .then(() => {
+        if (this._finishedSpans.length >= this._maxExportBatchSize) {
+          this._flushOneBatchAndContinue()
+        } else if (this._finishedSpans.length > 0) {
+          this._clearTimer();
+          this._maybeStartTimer();
+        }
+      })
+      .catch(e => {
+        globalErrorHandler(e);
+        if (this._finishedSpans.length >= this._maxExportBatchSize) {
+          this._flushOneBatchAndContinue()
```

Comment:
can't this cause endless recursion if you keep failing to flush?

Reply:
That is indeed a problem if it fails before removing the spans from the _finishedSpans. I'll try to cover for that case.

Reply:
@NimrodAvni78 
I restructure the in such a way that the spans are removed from `_finishedSpans` before any possible failure point. Meaning that availability of spans will limit the number of iterations thus preventing an infinite loop.